File: app/src/main/java/me/ykrank/s1next/viewmodel/SearchUserViewModel.java
package me.ykrank.s1next.viewmodel;

import android.view.View;

import androidx.databinding.ObservableField;
import androidx.fragment.app.FragmentActivity;

import me.ykrank.s1next.data.api.model.search.UserSearchResult;
import me.ykrank.s1next.view.activity.UserHomeActivity;
import me.ykrank.s1next.widget.glide.AvatarFailUrlsCache;


public final class SearchUserViewModel {

    public final ObservableField<UserSearchResult> search = new ObservableField<>();

    public void onClick(View v, View avatarView) {
        //Clear avatar false cache
        AvatarFailUrlsCache.clearUserAvatarCache(search.get().getUid());
        //个人主页
        UserHomeActivity.Companion.start((FragmentActivity) v.getContext(), search.get().getUid(), search.get().getName(), avatarView);
    }
}

File: app/src/main/java/me/ykrank/s1next/viewmodel/SearchUserViewModel.kt
package me.ykrank.s1next.viewmodel

import android.view.View
import androidx.databinding.ObservableField
import androidx.fragment.app.FragmentActivity
import me.ykrank.s1next.data.api.model.search.UserSearchResult
import me.ykrank.s1next.view.activity.UserHomeActivity.Companion.start

class SearchUserViewModel {
    val search = ObservableField<UserSearchResult>()
    fun onClick(v: View, avatarView: View) {
        //个人主页
        search.get()?.uid?.apply {
            start(
                (v.context as FragmentActivity),
                this,
                search.get()?.name,
                avatarView
            )
        }
    }
}
